File: go+vue/src/server/storage/files.go
package storage

import (
	"fmt"
	"log"
	"os"
)

// GetRecipeImage returns recipe image from file storage
func GetRecipeImage(recipeID int64) (*os.File, error) {
	filePath := fmt.Sprintf("%s%d", imagesFolder, recipeID)
	file, err := os.OpenFile(filePath, os.O_RDONLY, 0666)

	if err != nil {
		log.Println(err)
		if os.IsNotExist(err) {
			return nil, ErrNotFound
		}

		return nil, err
	}

	return file, nil
}

// AddRecipeImage adds recipe image to file storage
func AddRecipeImage(recipeID int64) (*os.File, error) {
	imagePath := fmt.Sprintf("%s%d", imagesFolder, recipeID)
	image, err := os.OpenFile(imagePath, os.O_WRONLY|os.O_CREATE, 0666)

	return image, err
}

// DeleteRecipeImage deletes recipe image from file storage
func DeleteRecipeImage(recipeID int64) error {
	filePath := fmt.Sprintf("%s%d", imagesFolder, recipeID)
	err := os.Remove(filePath)

	return err
}

File: noscript/src/server/storage/files.js
const path = require('path');
const fs = require('fs');
const config = require('../../../config');
const checkFileExists = require('../utils/check-file-exists');
const removeFile = require('../utils/remove-file');

const imageFolder = path.resolve(
  __dirname,
  '../../../',
  config.server.imageFolder
);

/**
 * Gets recipe image stream
 *
 * @param {number} recipeId
 * @return {Promise.<stream|null>} image file stream, otherwise file not found
 */
async function getRecipeImage(recipeId) {
  const filePath = `${imageFolder}/${recipeId}`;
  const exists = await checkFileExists(filePath);
  if (exists) {
    return fs.createReadStream(filePath);
  }
  return null;
}

/**
 * Adds recipe image
 *
 * @param {number} recipeId
 * @param {stream} fileStream
 * @return {Promise}
 */
function addRecipeImage(recipeId, fileStream) {
  return new Promise(res => {
    const filePath = `${imageFolder}/${recipeId}`;
    const w = fs.createWriteStream(filePath);
    fileStream.pipe(w);
    w.on('close', () => res());
  });
}

/**
 * Deletes recipe image
 *
 * @param {number} recipeId
 * @return {Promise.<boolean>}
 */
async function deleteRecipeImage(recipeId) {
  const filePath = `${imageFolder}/${recipeId}`;
  const exists = await checkFileExists(filePath);
  if (exists) {
    await removeFile(filePath);
  }
  return exists;
}

module.exports = {
  getRecipeImage,
  addRecipeImage,
  deleteRecipeImage
};
